File: lib/java/src/main/java/uapi/internal/types/GetType.java
package uapi.internal.types;

import java.util.List;
import java.util.Map;

public class GetType {

    public static String getType(Object value) {
        if (value == null) {
            return "Null";
        } else if (value instanceof Boolean) {
            return "Boolean";
        } else if (value instanceof Number) {
            return "Number";
        } else if (value instanceof String) {
            return "String";
        } else if (value instanceof List) {
            return "Array";
        } else if (value instanceof Map) {
            return "Object";
        } else {
            return "Unknown";
        }
    }
}

File: lib/py/uapi/internal/types/GetType.py
from typing import object, list, dict


def get_type(value: object) -> str:
    if value is None:
        return "Null"
    elif isinstance(value, bool):
        return "Boolean"
    elif isinstance(value, (int, float)):
        return "Number"
    elif isinstance(value, str):
        return "String"
    elif isinstance(value, list):
        return "Array"
    elif isinstance(value, dict):
        return "Object"
    else:
        return "Unknown"
